### PR TITLE
#1039: Added base_url variable to maestro notification twig

### PR DESCRIPTION
**Pull request by jekuaitk**
* Adds `base_url` variable to maestro notification twig
* Replaces tokens in all html rendered

Currently (with our docker setup) the stylesheet cannot be found if overwriting templates as seen in https://github.com/itk-dev/os2forms_selvbetjening/pull/291.

### CODE REVIEW

File: modules/os2forms_attachment/os2forms_attachment.module [hunk 1]
```diff
@@ -317,3 +318,12 @@ function os2forms_attachment_module_implements_alter(&$implementations, $hook) {
...
 
 }
+
+/**
+ * Implements hook_preprocess().
+ *
+ * Add 'base_url' variable to be used by templates.
+ */
+function os2forms_attachment_preprocess(&$vars) {
+  $vars['base_url'] = Settings::get('base_url');
```
Comment (rimi-itk) [inline modules/os2forms_attachment/os2forms_attachment.module:328]: We need some documentation on how and where to set the actual `base_url` value.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: modules/os2forms_attachment/os2forms_attachment.module (hunk 1 above)
Comment (jekuaitk) [reply modules/os2forms_attachment/os2forms_attachment.module:328]: Added documentation.

File: CHANGELOG.md [hunk 2]
```diff
@@ -14,6 +14,9 @@ before starting to add changes. Use example [placed in the end of the page](#exa
 - Adding Lat and Long fetching to DataAddress
 - [#84](https://github.com/OS2Forms/os2forms/pull/84)
   Added digital post test command.
+- [#95](https://github.com/OS2Forms/os2forms/pull/95)
+  - Added `base_url` variable to twig templates.
+  - Tokenized all Maestro notification html.
```
Comment (rimi-itk) [inline CHANGELOG.md:19]:
```suggestion
  - Handled tokens in Maestro notification html.
```

may be a more appropriate phrasing.

File: modules/os2forms_attachment/README.md [hunk 3]
```diff
@@ -13,3 +13,23 @@ To add custom headers/footer ```admin/structure/webform/config/os2forms_attachme
 To specify headers/footers that will override the default ones on a global level (**Third party settings** -> **Entity print** section): ```admin/structure/webform/config```
 
 To specify headers/footers that will override the default ones on a form level (**Third party settings** -> **Entity print** section): ```/admin/structure/webform/manage/[webform]/settings```
+
```
Comment (rimi-itk) [inline modules/os2forms_attachment/README.md:16]: The coding standards should complain about this superfluous blank line

```suggestion
```

File: modules/os2forms_forloeb/README.md [hunk 4]
```diff
@@ -44,3 +44,19 @@ must be processed asynchronously. Specify the queue handling notification jobs.
 #### Templates
 
 Define templates for emails and digital post (PDF).
+
+With some setups it might be necessary to import stylesheets with a full path.
+
+For this reason the `base_url` variable has been added for use in templates.
+Set it in `settings.local.php`
```
Comment (rimi-itk) [inline modules/os2forms_forloeb/README.md:51]: It's not a full "path", but a “URL”. Consider rewording to something like

> To reference assets, e.g. stylesheet or images, in your templates, you can use the `base_url` Twig variable to get the base URL:
> 
> ```twig
>   <link rel="stylesheet" href="{{ base_url }}/sites/default/templates/notification.html.twig
> ```
> 
> The value of `base_url` is defined in `settings.local.php`:

**Review by rimi-itk — CHANGES_REQUESTED** [review]

**Review by rimi-itk — APPROVED** [review]

**Review by thollebellcom — APPROVED** [review]

**Pull request merged**